ipykernel/ipkernel.py

```diff
@@ -124,6 +121,31 @@ def set_parent(self, ident, parent):
         super(IPythonKernel, self).set_parent(ident, parent)
         self.shell.set_parent(parent)
 
+    def init_metadata(self, parent):
+        """Initialize metadata.
+        
+        Run at the beginning of request handlers.
+        """
+        md = super(IPythonKernel, self).init_metadata(parent)
+        # FIXME: remove ipyparallel-specific code
```

Comment:
Going forward, these are provided by a subclass in ipyparallel. Removing these would be a breaking change, and would break ipyparallel < 5. We can consider dropping then in ipykernel 5 or 6.